docs/source/index.rst

```diff
@@ -0,0 +1,275 @@
+        py_modules=['example_plugin'],
+        entry_points={'example': {'example_plugin = example_plugin:ExamplePlugin'}}
+    )
+
+Now you can install the plugin running in console::
+
+    $ pip install -e .
+
+Current plugin
+--------------
+
+Very useful feature of Flask-PluginEngine is a ``current_plugin`` global value. It's a proxy to currently active plugin.
+
+Example plugin functionality
+----------------------------
+
+Now let's give our plugin some functionality to show how you can use the plugin. In this example we will render plugin's template from our application.
+
+Let's create the templates folder with a simple Jinja template. For example displaying plugin's title and description::
+
+    {% extends 'base.html' %}
+
+    {% block content %}
+        <h1>{{ plugin.title }}</h1>
+        <div>{{ plugin.description }}</div>
+    {% endblock %}
+
+If you documented plugin's class now you can access it as a ``name`` and ``description`` class properties.
+Notice that the template extends a base.html. We are going to use application's base template.
+
+Let's create a blueprint in our plugin::
+
+    plugin_blueprint = PluginBlueprint('example', __name__)
+
+In order to create a blueprint of a plugin we need to create PluginBlueprint instance.
+
+Then we can create a function rendering the template::
+
+    @plugin_blueprint.route('/plugin')
+    def hello_plugin():
+    return render_template('example_plugin:index.html', plugin=current_plugin)
+
+We can render any plugin template inside our application specifying name of the plugin before template name in ``render_template`` function, like above.
+We are also passing a current_plugin object to the template.
+
+
+Now in the plugin class we can create a method returning plugin's blueprint::
+
+    def get_blueprint(self):
+        return plugin_blueprint
+
+So our plugin code should look like this::
+
+    from flask import render_template
+    from flask_pluginengine import Plugin, PluginBlueprint, current_plugin
+
+    plugin_blueprint = PluginBlueprint('example', __name__)
+
+
+    @plugin_blueprint.route('/plugin')
+    def hello_plugin():
+        return render_template('example_plugin:index.html', plugin=current_plugin)
+
+
+    class ExamplePlugin(Plugin):
+        """ExamplePlugin
+
+        Just an example plugin
+        """
+
+        def get_blueprint(self):
+            return plugin_blueprint
+
+
+To get more about information see `Plugin`_ API reference.
+
+Minimal Application set up
+--------------------------
+
+To create Flask application using Flask-PluginEngine we need to define it as PluginFlask application::
+
+    from flask_pluginengine import PluginFlask
+    app = PluginFlask(__name__)
+
+    @app.route("/")
+    def hello():
+        return "Hello world!"
+
+    if __name__ == "__main__":
+        app.run()
+
+Now we need to set up the configuration: namespace of plugins we will be using with the application and plugins we want to use.
```

Comment:
I am a bit lost with the syntax but (if I understood it correctly) I think this is cleaner:
"Now we need to set up the configuration, **specifically the** namespace of **the** plugins ~we will be using with the application~ and **the** plugins we want to use."